### DISCUSSION TITLE
Bug Report: Export Option Not Working

### DISCUSSION THREAD

**Discussion by BenOussama180** [Power-Components/livewire-powergrid#499] [Q&A] [answered]
I get this Error While Exporting  my Orders Table : 
![image](https://user-images.githubusercontent.com/81111919/173707347-a8509e15-1ec3-4fbc-8bbb-1e7ed2eed98c.png)

This is my table : 
`<?php

namespace App\Http\Livewire;

use App\Models\Order;
use Illuminate\Support\Carbon;
use Illuminate\Database\QueryException;
use Illuminate\Database\Eloquent\Builder;
use PowerComponents\LivewirePowerGrid\Button;
use PowerComponents\LivewirePowerGrid\Column;
use PowerComponents\LivewirePowerGrid\PowerGrid;
use PowerComponents\LivewirePowerGrid\PowerGridEloquent;
use PowerComponents\LivewirePowerGrid\PowerGridComponent;
use PowerComponents\LivewirePowerGrid\Traits\ActionButton;
use PowerComponents\LivewirePowerGrid\Rules\Rule;
use PowerComponents\LivewirePowerGrid\{Header, Footer, Exportable};

final class OrderTable extends PowerGridComponent
{
    use ActionButton;

    //Messages informing success/error data is updated.
    public bool $showUpdateMessages = true;

    /*
    |--------------------------------------------------------------------------
    |  Features Setup
    |--------------------------------------------------------------------------
    | Setup Table's general features
    |
    */
    public function setUp(): array
    {
        return [
            Exportable::make('export')
                ->type(Exportable::TYPE_XLS, Exportable::TYPE_CSV),
            Header::make()
                ->showToggleColumns()
                ->showSearchInput(),
            Footer::make()
                ->showPerPage()
                ->showRecordCount('full'),
        ];
    }

    /*
    |--------------------------------------------------------------------------
    |  Datasource
    |--------------------------------------------------------------------------
    | Provides data to your Table using a Model or Collection
    |
    */

    /**
     * PowerGrid datasource.
     *
     * @return  \Illuminate\Database\Eloquent\Builder<\App\Models\Order>|null
     */
    public function datasource(): ?Builder
    {
        return Order::query()->groupBy('customer_id');
    }

    /*
    |--------------------------------------------------------------------------
    |  Relationship Search
    |--------------------------------------------------------------------------
    | Configure here relationships to be used by the Search and Table Filters.
    |
    */

    /**
     * Relationship search.
     *
     * @return array<string, array<int, string>>
     */
    public function relationSearch(): array
    {
        return [];
    }

    /*
    |--------------------------------------------------------------------------
    |  Add Column
    |--------------------------------------------------------------------------
    | Make Datasource fields available to be used as columns.
    | You can pass a closure to transform/modify the data.
    |
    */
    public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('id')
            ->addColumn('customer', function (Order $model) {
                return '<a class="underline" href="' . route('orders.customer', $model->customer_id) . '">' . $model->customer->email . '</a>';
            })
            ->addColumn('currency')
            ->addColumn('topic')
            ->addColumn('customer_cancel')
            ->addColumn('project_name')
            ->addColumn('created_at_formatted', function (Order $model) {
                return Carbon::parse($model->created_at)->format('d/m/Y H:i:s');
            })
            ->addColumn('pending', function (Order $model) {
                return '<span class="bg-red-100 text-red-800 text-sm font-medium mr-2 px-2.5 py-0.5 rounded dark:bg-green-200 dark:text-green-900">' . $model->customer->orders->where('status', "0")->count() . ' ' . 'NEW </span>';
            })
            ->addColumn('updated_at_formatted', function (Order $model) {
                return Carbon::parse($model->updated_at)->format('d/m/Y H:i:s');
            });
    }

    /*
    |--------------------------------------------------------------------------
    |  Include Columns
    |--------------------------------------------------------------------------
    | Include the columns added columns, making them visible on the Table.
    | Each column can be configured with properties, filters, actions...
    |
    */

    /**
     * PowerGrid Columns.
     *
     * @return array<int, Column>
     */
    public function columns(): array
    {
        return [
            Column::add()
                ->title('ID')
                ->field('id')
                ->sortable(),

            Column::add()
                ->title('CUSTOMER')
                ->field('customer')
                ->searchable(),

            Column::add()
                ->title('PENDING')
                ->field('pending')
                ->searchable(),

            Column::add()
                ->title('CREATED AT')
                ->field('created_at_formatted', 'created_at')
                ->searchable()
                ->sortable()
                ->makeInputDatePicker('created_at'),

        ];
    }`

**Comment by luanfreitasdev** [comment]
From the error you are giving, some field is returning an array right?

**Comment by BenOussama180** [comment]
well yes i think so but i cant tell which one .... @luanfreitasdev

**Comment by luanfreitasdev** [comment]
Hello.

I believe you will have to pass the data in string format in `addColumns` in the field that is returning array to work

**Comment by luanfreitasdev** [comment, marked as answer]
Fixed https://github.com/Power-Components/livewire-powergrid/releases/tag/v3.5.7
